web-wallet/src/lib/migration/migration.js

```diff
@@ -0,0 +1,165 @@
+import { readContract, simulateContract, writeContract } from "@wagmi/core";
+import { formatUnits, parseUnits } from "viem";
+import ERC20Abi from "./abi/erc_bep_20.json";
+import migrationABI from "./abi/migrationABI.json";
+import { wagmiConfig } from "./walletConnection";
+
+// Contract addresses for the DUSK tokens
+const duskContractAddress = "0x1e3364e8e93b5135645f49a1cd5b13fbdc02a19f";
+const stableCoinDecimals = 18;
+
+/**
+ * Retrieves the allowance amount for a stable coin for a given spender and account.
+ *
+ * @param {`0x${string}`} userAddress - The address of the user to check if it gave an allowance to the migration contract.
+ * @param {`0x${string}`} stableCoinAddress - contract address for the coin.
+ * @param {`0x${string}`} migrationContract - migration contract address for the the coin.
+ * @returns {Promise<string>} - A promise that resolves to the allowed amount.
+ * @throws {Error} - Throws an error if there is an issue with retrieving the allowance.
+ */
+export const allowance = async (
+  userAddress,
+  stableCoinAddress,
+  migrationContract
+) => {
+  try {
+    const balance = /** @type {bigint} */ (
+      await readContract(wagmiConfig, {
+        abi: ERC20Abi,
+        address: stableCoinAddress,
+        args: [userAddress, migrationContract],
+        functionName: "allowance",
+      })
+    );
+    return formatUnits(balance, stableCoinDecimals);
+  } catch (e) {
+    if (e instanceof Error) {
+      throw new Error(
+        `An error occurred while checking USDT spender: ${e.message}`
+      );
+    } else {
+      throw new Error(
+        `An unexpected error occurred while checking USDT spender: ${e}`
+      );
+    }
```

Comment:
I would do something like:

```
const errorMessage =
  e instanceof Error
    ? `An error occurred while checking USDT spender ${e.message}`
    : "An unknown error occurred while checking USDT spender";
  throw new Error(errorMessage);
}
```

Reply:
Same for the rest of the error handling in this file.

Obviously not a big deal, just makes the code a bit shorter.